### PR TITLE
[PM-8050] Deleting one folder deletes all other folders

### PR DESCRIPTION
**Pull request by gbubemismith**
## 🎟️ Tracking
[PM-8050](https://bitwarden.atlassian.net/browse/PM-8050)

## 📔 Objective
Deleting a folder should not clear all folders on the client. There seems to be a malfunction with `SyncService.syncDeleteFolder`, which tries to sync the web client after a folder has been deleted on the web client. Previously, we returned when `folders[id] == null`, which clears the folder state.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-8050]: https://bitwarden.atlassian.net/browse/PM-8050?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/web/src/app/vault/individual-vault/folder-add-edit.component.ts [hunk 1]
```diff
@@ -51,10 +52,12 @@ export class FolderAddEditComponent extends BaseFolderAddEditComponent {
     }
 
     try {
-      this.deletePromise = this.folderApiService.delete(this.folder.id);
-      await this.deletePromise;
-      this.platformUtilsService.showToast("success", null, this.i18nService.t("deletedFolder"));
-      this.onDeletedFolder.emit(this.folder);
```
Comment (shane-melton) [inline apps/web/src/app/vault/individual-vault/folder-add-edit.component.ts:57]: ❓ Did we intentionally drop this emission of `onDeletedFolder`?

File: libs/common/src/vault/services/folder/folder.service.ts [hunk 2]
```diff
@@ -137,16 +137,14 @@ export class FolderService implements InternalFolderServiceAbstraction {
         return;
       }
 
-      if (typeof id === "string") {
-        if (folders[id] == null) {
-          return;
```
Comment (shane-melton) [inline libs/common/src/vault/services/folder/folder.service.ts:142]: 👍 Wow, nice catch on this one!

**Review by shane-melton — COMMENTED** [review]
Nice catch on this bug! Just one unrelated question to the `onDeletedFolder` event emitter.

File: apps/web/src/app/vault/individual-vault/folder-add-edit.component.ts (hunk 1 above)
Comment (gbubemismith) [reply apps/web/src/app/vault/individual-vault/folder-add-edit.component.ts:57]: The vault component calls the component library dialog for this and doesn't need the emitted output anymore; I think when it was refactored, it wasn't cleaned up

Comment (shane-melton) [reply apps/web/src/app/vault/individual-vault/folder-add-edit.component.ts:57]: Understood, thanks for the clarification!

**Review by shane-melton — APPROVED** [review]

**Pull request merged**